packages/devtools_app/test/vm_service_private_test.dart

```diff
@@ -0,0 +1,43 @@
+import 'package:devtools_app/src/globals.dart';
+import 'package:devtools_app/src/service_manager.dart';
+import 'package:devtools_app/src/vm_service_wrapper.dart';
+import 'package:mockito/mockito.dart';
+import 'package:test/test.dart';
+import 'package:vm_service/vm_service.dart';
+
+import 'support/mocks.dart';
+
+void main() {
+  FakeServiceManager fakeServiceManager;
+
+  setUp(() {
+    fakeServiceManager = FakeServiceManager();
+    setGlobal(ServiceConnectionManager, fakeServiceManager);
+  });
+  test('Ensure private RPCs can only be enabled with VM Developer Mode enabled',
+      () async {
+    // ignore: unnecessary_cast
+    when(fakeServiceManager.service.trackFuture as Function).thenReturn(
+      <T>(String _, Future<T> future) => future,
+    );
+    when(fakeServiceManager.service
+            .callMethod(argThat(equals('_collectAllGarbage'))))
+        .thenAnswer(
+      (_) => Future.value(Success()),
+    );
```

Comment:
can we just override these methods directly in the FakeServiceManager class?

Reply:
It's not trivial to override `callMethod` since its response varies depending on the method being invoked. We'd have to add a `Response callMethodResult` property to handle this, but that wouldn't work for more complex tests where `callMethod` is called multiple times with different results within a method being tested which is why I opted to use a mock instead of a fake VM service.

However, I'm willing to switch to a fake if you'd prefer.

Reply:
Ah I see what you are saying. Leaving as is is fine.

Reply:
I was originally talking about mocking and overriding `VmServicePrivate.collectAllGarbage` but I see now why that wouldn't work